odl/tomo/backends/astra_cuda.py

```diff
@@ -0,0 +1,199 @@
+# Copyright 2014, 2015 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Backen for ASTRA using CUDA"""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+# External
+import numpy as np
+
+try:
+    import astra
+    if astra.astra.use_cuda():
+        ASTRA_CUDA_AVAILABLE = True
+except ImportError:
+    ASTRA_CUDA_AVAILABLE = False
+
+# Internal
+from odl.discr.lp_discr import DiscreteLp, DiscreteLpVector
+from odl.tomo.backends.astra_setup import (astra_projection_geometry,
+                                           astra_volume_geometry,
+                                           astra_projector, astra_data,
+                                           astra_algorithm, astra_cleanup)
+from odl.tomo.geometry.geometry import Geometry
+
+
+__all__ = ('astra_gpu_forward_projector_call',
+           'astra_gpu_backward_projector_call',
+           'ASTRA_CUDA_AVAILABLE')
+
+
+# TODO: rename gpu to cuda?
+def astra_gpu_forward_projector_call(vol_data, geometry, proj_space, out=None):
+    """Run an ASTRA forward projection on the given data using the GPU.
+
+    Parameters
+    ----------
+    vol_data : `DiscreteLpVector`
+        Volume data to which the projector is applied
```

Comment:
I followed the general rule that if it's just a one-line summary which does not make for a proper sentence (you know, subject, predicate[, object]), don't use punctuation. Otherwise, do (e.g. when there is a second sentence following). Punctuation is there to conclude proper sentences, so there is nothing unprofessional about leaving them out after half-sentences.